src/Workspaces/Core/Portable/Utilities/Contract.cs

```diff
@@ -23,6 +23,13 @@ internal static class Contract
             }
         }
 
+        [return: NotNull]
+        public static T NotNull<T>(T value, string? message = null) where T : class?
```

Comment:
btw `[NotNull]` on `value` in other methods is suspicious. Do we need it here too? (I think not)